README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

# sierraleone: Datasets for use in designing surveys in Sierra Leone

<!-- badges: start -->

[![Lifecycle:
maturing](https://img.shields.io/badge/lifecycle-maturing-blue.svg)](https://www.tidyverse.org/lifecycle/#maturing)
[![Travis build
status](https://travis-ci.org/spatialworks/sierraleone.svg?branch=master)](https://travis-ci.org/spatialworks/sierraleone)
[![AppVeyor build
status](https://ci.appveyor.com/api/projects/status/github/spatialworks/sierraleone?branch=master&svg=true)](https://ci.appveyor.com/project/spatialworks/sierraleone)
<!-- badges: end -->

Designing surveys require relevant datasets to be used as basis for
sample size calculations, sampling design, survey planning/logistics and
survey implementation. These include datasets on population, lists of
sampling clusters, map datasets for spatial sampling, and previous
survey datasets that can be used for estimating indicator variance and
design effects. This package contains relevant datasets for use in
designing surveys in Sierra Leone.

## Installation

The `sierraleone` packages is not yet released on
[CRAN](https://cran.r-project.org). The development version of the
package can be installed from
[GitHub](https://github.com/spatialworks/sierraleone) with:

``` r
if(!require(remotes)) install.packages("remotes")
remotes::install_github("spatialworks/sierraleone")
```

## Usage

When installing `sierraleone`, geospatial packages on which
`sierraleone` depends on are installed. To use `sierraleone` package, it
will be important to load these package dependencies that have been
installed. This can be done by:

``` r
library(rgdal)
library(rgeos)
library(raster)
```

### Province borders

The Sierra Leone province borders `SpatialPolygonsDataFrame` is accessed
via the `province` dataset.

``` r
sierraleone::province
#> class       : SpatialPolygonsDataFrame 
#> features    : 4 
#> extent      : -13.30901, -10.27056, 6.923379, 9.999253  (xmin, xmax, ymin, ymax)
#> crs         : +proj=longlat +datum=WGS84 +no_defs +ellps=WGS84 +towgs84=0,0,0 
#> variables   : 13
#> names       : OBJECTID, admin1Name, admin1Pcod, admin1RefN, admin1AltN, admin1Al_1,   admin0Name, admin0Pcod,       date,    validOn, validTo,   Shape_Leng,      Shape_Area 
#> min values  :        1,    Eastern,       SL01,    Eastern,         NA,         NA, Sierra Leone,         SL, 2016/08/01, 2016/10/17,      NA, 1.8390606275, 0.0563871543411 
#> max values  :        4,    Western,       SL04,    Western,         NA,         NA, Sierra Leone,         SL, 2016/08/01, 2016/10/17,      NA, 12.645889079,   2.96657963229
```

The province borders of Sierra Leone can be plotted by:

``` r
sp::plot(sierraleone::province)
```

<img src="man/figures/README-provincePlot-1.png" width="100%" />

### District borders

The Sierra Leone district borders `SpatialPolygonsDataFrame` is accessed
via the `district` dataset.

``` r
sierraleone::district
#> class       : SpatialPolygonsDataFrame 
#> features    : 14 
#> extent      : -13.30901, -10.27056, 6.923379, 9.999253  (xmin, xmax, ymin, ymax)
#> crs         : +proj=longlat +datum=WGS84 +no_defs +ellps=WGS84 +towgs84=0,0,0 
#> variables   : 15
#> names       : OBJECTID,         admin2Name, admin2Pcod,         admin2RefN, admin2AltN, admin2Al_1, admin1Name, admin1Pcod,   admin0Name, admin0Pcod,       date,    validOn, ValidTo,     Shape_Leng,       Shape_Area 
#> min values  :        1,                 Bo,     SL0101,                 Bo,         NA,         NA,    Eastern,       SL01, Sierra Leone,         SL, 2016/08/01, 2016/10/17,      NA, 0.738326560804, 0.00606880539992 
#> max values  :        9, Western Area Urban,     SL0402, Western Area Urban,         NA,         NA,    Western,       SL04, Sierra Leone,         SL, 2016/08/01, 2016/10/17,      NA,  6.84446934587,    1.01774343267
```

The district borders of Sierra Leone can be plotted by:

``` r
sp::plot(sierraleone::district)
```

<img src="man/figures/README-districtPlot-1.png" width="100%" />

### Chiefdom borders

The Sierra Leone chiefdom borders `SpatialPolygonsDataFrame` is accessed
via the `chiefdom` dataset.

``` r
sierraleone::chiefdom
#> class       : SpatialPolygonsDataFrame 
#> features    : 167 
#> extent      : -13.30901, -10.27056, 6.923379, 9.999253  (xmin, xmax, ymin, ymax)
#> crs         : +proj=longlat +datum=WGS84 +no_defs +ellps=WGS84 +towgs84=0,0,0 
#> variables   : 19
#> names       : OBJECTID, admin3Name, admin3Pcod, admin3RefN,         admin2Name, admin2Pcod, admin1Name, admin1Pcod,   admin0Name, admin0Pcod,       date,    validOn, validTo,      Shape_Leng,        Shape_Area, ... 
#> min values  :        1,     Badjia,   SL010101,     Badjia,                 Bo,     SL0101,    Eastern,       SL01, Sierra Leone,         SL, 2016/08/01, 2016/10/17,      NA, 0.0487843097851, 6.51303509621e-05, ... 
#> max values  :       99, York Rural,   SL040208, York Rural, Western Area Urban,     SL0402,    Western,       SL04, Sierra Leone,         SL, 2016/08/01, 2016/10/17,      NA,   2.76373793498,    0.202641459488, ...
```

The chiefdom borders of Sierra Leone can be plotted by:

``` r
sp::plot(sierraleone::chiefdom)
```

<img src="man/figures/README-chiefdomPlot-1.png" width="100%" />

### County borders

The Sierra Leone county borders `SpatialPolygonsDataFrame` is accessed
via the `county` dataset.

``` r
sierraleone::county
#> class       : SpatialPolygonsDataFrame 
#> features    : 1316 
#> extent      : -13.30901, -10.27056, 6.923379, 9.999253  (xmin, xmax, ymin, ymax)
#> crs         : +proj=longlat +datum=WGS84 +no_defs +ellps=WGS84 +towgs84=0,0,0 
#> variables   : 24
#> names       : OBJECTID, admin4Name, admin4Pcod, admin4RefN, admin4AltN, admin4Al_1, admin3Name, admin3Pcod,         admin2Name, admin2Pcod, admin1Name, admin1Pcod,   admin0Name, admin0Pcod,       date, ... 
#> min values  :        1,   Aberdeen, SL01010101,   Aberdeen,         NA,         NA,     Badjia,   SL010101,                 Bo,     SL0101,    Eastern,       SL01, Sierra Leone,         SL, 2016/08/01, ... 
#> max values  :      999,     Zone 7, SL04020810,     Zone 7,         NA,         NA, York Rural,   SL040208, Western Area Urban,     SL0402,    Western,       SL04, Sierra Leone,         SL, 2016/08/01, ...
```

The county borders of Sierra Leone can be plotted by:

``` r
sp::plot(sierraleone::county)
```

<img src="man/figures/README-countyPlot-1.png" width="100%" />


NEWS.md
# sierraleone 0.1.0

* Added a `NEWS.md` file to track changes to the package.


data-raw/processData.R
## Libraries
library(rgdal)
library(rgeos)
library(raster)

##
options(stringsAsFactors = FALSE)

##
link1 <- "https://data.humdata.org/dataset/a4816317-a913-4619-b1e9-d89e21c056b4/resource/b6225ff4-80e2-428f-be5e-db5d2f43a3d6/download/sle_admbnda_adm1_1m_gov_ocha_20161017.shp.zip"
link2 <- "https://data.humdata.org/dataset/a4816317-a913-4619-b1e9-d89e21c056b4/resource/b3963917-8550-478d-9363-736492bf209a/download/sle_admbnda_adm2_1m_gov_ocha.zip"
link3 <- "https://data.humdata.org/dataset/a4816317-a913-4619-b1e9-d89e21c056b4/resource/e2aa661d-af2f-42d8-bdea-c7e16a00bdb2/download/sle_admbnda_adm3_1m_gov_ocha_20161017.zip"
link4 <- "https://data.humdata.org/dataset/a4816317-a913-4619-b1e9-d89e21c056b4/resource/da852c80-d459-4514-a835-d1d0d457ef2a/download/sle_admbnda_adm4_1m_gov_ocha.zip"

## Province
temp <- tempfile()
download.file(url = link1, destfile = temp)

unzip(zipfile = temp, exdir = tempdir())

x <- list.files(tempdir())[stringr::str_detect(string = list.files(tempdir()), pattern = "adm1")]
layer <- stringr::str_remove(string = x[1], pattern = ".cpg|.dbf|.prj|.sbn|.sbx|.shp|.shp.xml|.shx")

province <- rgdal::readOGR(dsn = tempdir(), layer = layer)

usethis::use_data(province, compress = "xz", overwrite = TRUE)

unlink(temp, recursive = TRUE)


## Districts
temp <- tempfile()
download.file(url = link2, destfile = temp)
unzip(zipfile = temp, exdir = tempdir())

x <- list.files(tempdir())[stringr::str_detect(string = list.files(tempdir()), pattern = "adm2")]
layer <- stringr::str_remove(string = x[1], pattern = ".cpg|.dbf|.prj|.sbn|.sbx|.shp|.shp.xml|.shx")

district <- rgdal::readOGR(dsn = tempdir(), layer = layer)

usethis::use_data(district, compress = "xz", overwrite = TRUE)

unlink(temp, recursive = TRUE)


## Chiefdoms
temp <- tempfile()
download.file(url = link3, destfile = temp)
unzip(zipfile = temp, exdir = tempdir())

x <- list.files(tempdir())[stringr::str_detect(string = list.files(tempdir()), pattern = "adm3")]
layer <- stringr::str_remove(string = x[1], pattern = ".cpg|.dbf|.prj|.sbn|.sbx|.shp|.shp.xml|.shx")

chiefdom <- rgdal::readOGR(dsn = tempdir(), layer = layer)

usethis::use_data(chiefdom, compress = "xz", overwrite = TRUE)

unlink(temp, recursive = TRUE)


## Counties
temp <- tempfile()
download.file(url = link4, destfile = temp)
unzip(zipfile = temp, exdir = tempdir())

x <- list.files(tempdir())[stringr::str_detect(string = list.files(tempdir()), pattern = "adm4")]
layer <- stringr::str_remove(string = x[1], pattern = ".cpg|.dbf|.prj|.sbn|.sbx|.shp|.shp.xml|.shx")

county <- rgdal::readOGR(dsn = tempdir(), layer = layer)

usethis::use_data(county, compress = "xz", overwrite = TRUE)

unlink(temp, recursive = TRUE)





R/data.R
################################################################################
#
#' Sierra Leone administrative area level 1 - province
#'
#' @format A SpatialPolygonsDataFrame with 3 features:
#' \describe{
#'   \item{\code{OBJECTID}}{Object Identifier}
#'   \item{\code{admin1Name}}{Name of administrative area level 1}
#'   \item{\code{admin1Pcod}}{Postal code of administrative area level 1}
#'   \item{\code{admin1RefN}}{Reference of administrative area level 1}
#'   \item{\code{admin1AltN}}{Alternative name of administrative area level 1}
#'   \item{\code{admin1Al_1}}{Alternative name of administrative area level 1}
#'   \item{\code{admin0Name}}{Name of administrative area level 0}
#'   \item{\code{admin0Pcod}}{Postal code of administrative area level 0}
#'   \item{\code{date}}{Date}
#'   \item{\code{validOn}}{Valid on}
#'   \item{\code{validTo}}{Valid to}
#'   \item{\code{Shape_Leng}}{Shape length}
#'   \item{\code{Shape_Area}}{Shape area}
#' }
#'
#' @examples
#' province
#'
#' @source \url{https://data.humdata.org/dataset/sierra-leone-all-ad-min-level-boundaries}
#'
#
################################################################################
"province"


################################################################################
#
#' Sierra Leone administrative area level 2 - district
#'
#' @format A SpatialPolygonsDataFrame with 14 features:
#' \describe{
#'   \item{\code{OBJECTID}}{Object Identifier}
#'   \item{\code{admin2Name}}{Name of administrative area level 2}
#'   \item{\code{admin2Pcod}}{Postal code of administrative area level 2}
#'   \item{\code{admin2RefN}}{Reference of administrative area level 2}
#'   \item{\code{admin2AltN}}{Alternative name of administrative area level 2}
#'   \item{\code{admin2Al_1}}{Alternative name of administrative area level 2}
#'   \item{\code{admin1Name}}{Name of administrative area level 1}
#'   \item{\code{admin1Pcod}}{Postal code of administrative area level 1}
#'   \item{\code{admin1RefN}}{Reference of administrative area level 1}
#'   \item{\code{admin1AltN}}{Alternative name of administrative area level 1}
#'   \item{\code{admin1Al_1}}{Alternative name of administrative area level 1}
#'   \item{\code{admin0Name}}{Name of administrative area level 0}
#'   \item{\code{admin0Pcod}}{Postal code of administrative area level 0}
#'   \item{\code{date}}{Date}
#'   \item{\code{validOn}}{Valid on}
#'   \item{\code{validTo}}{Valid to}
#'   \item{\code{Shape_Leng}}{Shape length}
#'   \item{\code{Shape_Area}}{Shape area}
#' }
#'
#' @examples
#' district
#'
#' @source \url{https://data.humdata.org/dataset/sierra-leone-all-ad-min-level-boundaries}
#'
#
################################################################################
"district"


################################################################################
#
#' Sierra Leone administrative area level 3 - chiefdom
#'
#' @format A SpatialPolygonsDataFrame with 167 features:
#' \describe{
#'   \item{\code{OBJECTID}}{Object Identifier}
#'   \item{\code{admin3Name}}{Name of administrative area level 3}
#'   \item{\code{admin3Pcod}}{Postal code of administrative area level 3}
#'   \item{\code{admin3RefN}}{Reference of administrative area level 3}
#'   \item{\code{admin3AltN}}{Alternative name of administrative area level 3}
#'   \item{\code{admin3Al_1}}{Alternative name of administrative area level 3}
#'   \item{\code{admin2Name}}{Name of administrative area level 2}
#'   \item{\code{admin2Pcod}}{Postal code of administrative area level 2}
#'   \item{\code{admin2RefN}}{Reference of administrative area level 2}
#'   \item{\code{admin2AltN}}{Alternative name of administrative area level 2}
#'   \item{\code{admin2Al_1}}{Alternative name of administrative area level 2}
#'   \item{\code{admin1Name}}{Name of administrative area level 1}
#'   \item{\code{admin1Pcod}}{Postal code of administrative area level 1}
#'   \item{\code{admin1RefN}}{Reference of administrative area level 1}
#'   \item{\code{admin1AltN}}{Alternative name of administrative area level 1}
#'   \item{\code{admin1Al_1}}{Alternative name of administrative area level 1}
#'   \item{\code{admin0Name}}{Name of administrative area level 0}
#'   \item{\code{admin0Pcod}}{Postal code of administrative area level 0}
#'   \item{\code{date}}{Date}
#'   \item{\code{validOn}}{Valid on}
#'   \item{\code{validTo}}{Valid to}
#'   \item{\code{Shape_Leng}}{Shape length}
#'   \item{\code{Shape_Area}}{Shape area}
#' }
#'
#' @examples
#' chiefdom
#'
#' @source \url{https://data.humdata.org/dataset/sierra-leone-all-ad-min-level-boundaries}
#'
#
################################################################################
"chiefdom"


################################################################################
#
#' Sierra Leone administrative area level 4 - county
#'
#' @format A SpatialPolygonsDataFrame with 1316 features:
#' \describe{
#'   \item{\code{OBJECTID}}{Object Identifier}
#'   \item{\code{admin4Name}}{Name of administrative area level 4}
#'   \item{\code{admin4Pcod}}{Postal code of administrative area level 4}
#'   \item{\code{admin4RefN}}{Reference of administrative area level 4}
#'   \item{\code{admin4AltN}}{Alternative name of administrative area level 4}
#'   \item{\code{admin4Al_1}}{Alternative name of administrative area level 4}
#'   \item{\code{admin3Name}}{Name of administrative area level 3}
#'   \item{\code{admin3Pcod}}{Postal code of administrative area level 3}
#'   \item{\code{admin3RefN}}{Reference of administrative area level 3}
#'   \item{\code{admin3AltN}}{Alternative name of administrative area level 3}
#'   \item{\code{admin3Al_1}}{Alternative name of administrative area level 3}
#'   \item{\code{admin2Name}}{Name of administrative area level 2}
#'   \item{\code{admin2Pcod}}{Postal code of administrative area level 2}
#'   \item{\code{admin2RefN}}{Reference of administrative area level 2}
#'   \item{\code{admin2AltN}}{Alternative name of administrative area level 2}
#'   \item{\code{admin2Al_1}}{Alternative name of administrative area level 2}
#'   \item{\code{admin1Name}}{Name of administrative area level 1}
#'   \item{\code{admin1Pcod}}{Postal code of administrative area level 1}
#'   \item{\code{admin1RefN}}{Reference of administrative area level 1}
#'   \item{\code{admin1AltN}}{Alternative name of administrative area level 1}
#'   \item{\code{admin1Al_1}}{Alternative name of administrative area level 1}
#'   \item{\code{admin0Name}}{Name of administrative area level 0}
#'   \item{\code{admin0Pcod}}{Postal code of administrative area level 0}
#'   \item{\code{date}}{Date}
#'   \item{\code{validOn}}{Valid on}
#'   \item{\code{validTo}}{Valid to}
#'   \item{\code{Shape_Leng}}{Shape length}
#'   \item{\code{Shape_Area}}{Shape area}
#' }
#'
#' @examples
#' county
#'
#' @source \url{https://data.humdata.org/dataset/sierra-leone-all-ad-min-level-boundaries}
#'
#
################################################################################
"county"
